agents/README.md

```diff
@@ -0,0 +1,223 @@
+4. That cluster has a namespace "rl" within which we will be running our training jobs.
+
+###### Creating a logs bucket
+
+Credentials are needed in order to make the calls to the Google Cloud platform APIs dispatched in subsequent steps. You can start by authenticating with GCloud in the usual way (i.e. `gcloud auth login` followed by `gcloud config set project <your-project-name>`).
+
+First, we need to create a Google Cloud Storage bucket to store job logs as well as a unique subdirectory of that bucket to store logs for this particular run. We can create that as follows:
+
+```bash
+gsutil mb -p <your-project-name> gs://<your-project-name>
+```
+
+Note that you'll need to set the path to this bucket at the beginning of the [demo notebook](demo/demo.ipynb) so remember it for later!
+
+###### Deploying Kubernetes
+
+If you haven't already you can deploy a Kubernetes cluster on Google Kubernetes Engine that will support Kubeflow with the following command:
+
+```bash
+# Your project here
+PROJECT=kubeflow-rl
+CLUSTER=kubeflow-dev
+ZONE=us-east1-d
+
+gcloud beta container clusters create \
+       $CLUSTER \
+       --project ${PROJECT} \
+       --zone=$ZONE \
+       --cluster-version=1.9.2-gke.1 \
+       --machine-type=n1-highmem-32 \
+       --image-type=COS \
+       --disk-size=100 \
+       --num-nodes=3 \
+       --enable-autoscaling \
+       --max-nodes=8 \
+       --min-nodes=1 \
+       --scopes=https://www.googleapis.com/auth/cloud-platform
+```
+
+We'll pull the Kubernetes credentials for that cluster to the machine we're using for setup to permit us to perform the subsequent steps of creating a service account secret and a namespace for our jobs.
+
+```bash
+gcloud container clusters --project=${PROJECT} \
+  --zone=${ZONE} get-credentials ${CLUSTER}
+```
+
+The rest of this tutorial assumes the cluster for which you have pulled credentials already has Kubeflow deployed. The steps to do so are described [here](https://github.com/kubeflow/kubeflow#steps); see also the [Kubeflow User Guide](https://github.com/kubeflow/kubeflow/blob/master/user_guide.md).
+
+**Note:** Beyond the core dependencies this demo depends on also having [Argo](https://applatix.com/open-source/argo/) (a Kubernetes-native workflow manager) deployed which for an existing Kubeflow deployment can be added as follows:
+
+```bash
+# From the root of your ksonnet workspace
+ks pkg install kubeflow/argo
+ks generate argo kubeflow-argo --name=kubeflow-argo --namespace=kubeflow
+ks apply default -c kubeflow-argo
+```
+
+Note that the above assumes you've run the `ks env set default` command after pulling your Kubernetes credentials.
+
+Note also that the above step assumes you have already performed the setup steps in the [Kubeflow user guide](https://github.com/kubeflow/kubeflow/blob/master/user_guide.md) including creating a Kubeflow namespace and creating an admin cluster role for the user deploying Argo, both of which can be done as follows:
```

Comment:
Can we move this into the user guide and explain how to deploy Argo in the user guide?